supriya/tools/ugentools/Duty.py

```diff
@@ -0,0 +1,265 @@
+# -*- encoding: utf-8 -*-
+from supriya.tools.ugentools.UGen import UGen
+
+
+class Duty(UGen):
+    r"""
+    A value is demanded of each UGen in the list and output according to a stream of duration values.
+
+    ::
+        >>> duty = ugentools.Duty.kr(
+        ...     done_action=0,
+        ...     duration=ugentools.Drand(
+        ...         sequence=[0.01, 0.2, 0.4],
+        ...         repeats=2,
+        ...     ),
+        ...     reset=0,
+        ...     level=ugentools.Dseq(
+        ...         sequence=[204, 400, 201, 502, 300, 200],
+        ...         repeats=2,
+        ...         ),
+        ...     )
+        >>> duty
+        Duty.kr()
+
+    """
+
+    # ### CLASS VARIABLES ###
+
+    __documentation_section__ = None
+
+    __slots__ = ()
+
+    _ordered_input_names = (
+        'duration',
+        'reset',
+        'level',
+        'done_action',
+    )
+
+    _valid_calculation_rates = None
+
+    # ### INITIALIZER ###
+
+    def __init__(
+            self,
+            calculation_rate=None,
+            done_action=0,
+            duration=1,
+            level=1,
+            reset=0,
+    ):
```

Comment:
While I'd prefer a hanging indent (this line indented four more spaces), I'm OK with this.

In the future, I may try to run Google's `yapf` Python formatter against the codebase. That code beautifier can't do the hanging-indent style I prefer, and would output this de-dented parens style instead. Oh well...